File: src/main/java/ru/citeck/ecos/apps/domain/EcosAppEntity.java
package ru.citeck.ecos.apps.domain;

import lombok.Getter;
import lombok.Setter;
import ru.citeck.ecos.apps.app.PublishStatus;

import javax.persistence.*;
import java.util.Set;

@Entity
@Table(name = "ecos_app")
public class EcosAppEntity extends AbstractAuditingEntity {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "ecos_app_id_gen")
    @SequenceGenerator(name = "ecos_app_id_gen")
    @Getter @Setter private Long id;

    @Column(name = "ext_id")
    @Getter @Setter private String extId;
    @Getter @Setter private String version;

    @Column(name = "is_system")
    @Getter @Setter private Boolean isSystem;

    @OneToMany(mappedBy = "application", fetch = FetchType.LAZY)
    @Getter @Setter private Set<EcosAppRevEntity> revisions;

    @Enumerated(EnumType.ORDINAL)
    @Column(name = "publish_status")
    @Getter @Setter private PublishStatus publishStatus = PublishStatus.DRAFT;
}

File: src/main/java/ru/citeck/ecos/apps/domain/ecosapp/repo/EcosAppEntity.kt
package ru.citeck.ecos.apps.domain.ecosapp.repo

import jakarta.persistence.*
import ru.citeck.ecos.apps.domain.content.repo.EcosContentEntity
import ru.citeck.ecos.webapp.lib.spring.hibernate.entity.AbstractAuditingEntity
import java.time.Instant

@Entity
@Table(name = "ecos_app")
class EcosAppEntity : AbstractAuditingEntity() {

    @Id
    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "hibernate_sequence")
    @SequenceGenerator(name = "hibernate_sequence")
    private val id: Long? = null

    override fun getId() = id
    lateinit var extId: String

    var name: String? = null
    var version: String? = null

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "artifacts_dir_content_id")
    var artifactsDir: EcosContentEntity? = null

    var artifactsLastModifiedDate: Instant? = null
}
